src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -346,27 +315,19 @@ public override void Write(byte[] buffer, int offset, int count)
                     throw new TimeoutException(BrotliEx.TimeoutWrite);
                 }
                 copyLen = bytesRemain > _bufferSize ? _bufferSize : bytesRemain;
-                Marshal.Copy(buffer, currentOffset, _bufferInput, copyLen);
+                byte[] bufferInput = new byte[copyLen];
```

Comment:
Not necessary It is temporary. What the problem is that signature of Compress takes out parameter (consumed) and input is exactly as source size. I can change consumed to ref parameter, but it is inconsistent with other Encode/Decode methods, which using Span